File: Algorithm/Go/932. Beautiful Array.go
func beautifulArray(N int) []int {
	result := []int{1}
	for len(result) < N {
		odds, evens := []int{}, []int{}
		for _, num := range result {
			if num*2-1 <= N {
				odds = append(odds, num*2-1)
			}
			if num*2 <= N {
				evens = append(evens, num*2)
			}
		}
		odds = append(odds, evens...)
		result = odds
	}
	return result
}

File: Algorithm/C++/932. Beautiful Array.cc
#include <vector>

class Solution {
 public:
    std::vector<int> beautifulArray(int N) {
        std::vector<int> result{1};
        while(result.size() < N) {
            std::vector<int> odds, evens;
            for(auto num: result) {
                if (num*2 - 1 <= N) odds.push_back(num*2-1);
                if (num*2 <= N) evens.push_back(num*2);
            }
            odds.insert(odds.end(), evens.begin(), evens.end());
            result=odds;
        }
        return result;
    }
};